lua/toggleterm/terminal.lua

```diff
@@ -121,9 +132,11 @@ local function setup_buffer_autocommands(term)
   local conf = config.get()
   api.nvim_create_augroup(AUGROUP, { clear = true })
   api.nvim_create_autocmd("TermClose", {
-    buffer = term.bufnr,
```

Comment:
If this isn't local to a buffer then it should use the pattern for toggleterm buffers, you can find this where the other terminal autocommands are. It's something like `pattern = 'term://*toggleterm*'`

Reply:
It probably should also not be set in a `setup_buffer_autocommands` because this will recreate this autocommand every time this function is called, which is every time a toggleterm is created.